### PR TITLE
DATA-734 SLAM Flaky Test

### PR DESCRIPTION
**Pull request by jeremyhyde-viam**
Deferred release of video stream had a chance to be missed due to error in image call, moved up a few lines and introduced an error check.

### CODE REVIEW

File: services/slam/builtin/builtin.go [hunk 1]
```diff
@@ -847,15 +849,20 @@ func (slamSvc *builtIn) getAndSaveDataSparse(
 		}
 
 		images, releaseFuncs, err := slamSvc.getSimultaneousColorAndDepth(ctx, cams)
+		if releaseFuncs[0] != nil {
```
Comment (edaniels) [inline services/slam/builtin/builtin.go:852]: do this in a loop with a var copy

File: services/slam/builtin/builtin.go [hunk 2]
```diff
@@ -847,15 +849,20 @@ func (slamSvc *builtIn) getAndSaveDataSparse(
...
+			defer releaseFuncs[0]()
+		}
+		if releaseFuncs[1] != nil {
+			defer releaseFuncs[0]()
```
Comment (edaniels) [inline services/slam/builtin/builtin.go:856]: loop gets rid of this but this is wrong func

File: services/slam/builtin/builtin.go [hunk 3]
```diff
@@ -815,14 +815,16 @@ func (slamSvc *builtIn) getAndSaveDataSparse(
 		}
 
 		image, release, err := slamSvc.getPNGImage(ctx, cams[0])
+		if release != nil {
```
Comment (edaniels) [inline services/slam/builtin/builtin.go:818]: how did you test this was the issue?

Comment (jeremyhyde-viam) [reply services/slam/builtin/builtin.go:818]: I ran tests of 10,000 trials: `go test . -run TestOrbslamYAMLNew -race -count 10000`

**Review by kkufieta — APPROVED** [review]
LGTM

**Review by JohnN193 — APPROVED** [review]
LGTM

File: services/slam/builtin/builtin.go (hunk 3 above)
Comment (edaniels) [reply services/slam/builtin/builtin.go:818]: coo

**Pull request merged**